index.js

```diff
@@ -46,7 +46,7 @@ function noir (serializers, keys, censor) {
   return shape
 
   function redact () { return {toJSON: mask} }
-  function mask () { return censor }
+  function mask () { return typeof censor === 'function' ? censor() : censor }
```

Comment:
the point of `censor` as a function is to take a value and then return a mask based on that value

no value is being passed to `censor` here.. (which could cause user issues as well, if they're expecting a value)

Reply:
This is one of the parts I don't have a good understanding of. There is no argument here that I can use. Not sure how to get a value... Can you think of a use case that'll fail this implementation? If you look at my current tests, this only gets called when an empty object is passed. If I can get an understanding, I can create a failing test.

Reply:
`mask` becomes the value of `toJSON` for top level objects 

`toJSON` is passed the key, while `this` is the object itself, so you can do `censor(this[key])`

see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify